### PR TITLE
Introducing Security Analytics

### LINKED ISSUE #1221: [Blog] Overview of security analytics in OpenSearch
**Describe the blog post your would like to write**
How OpenSearch is used for security, including what industries and verticals choose OpenSearch and why. What functionality and purpose-built tools OpenSearch Project incorporates for security applications. Recent developments with the release of 2.4.0, and preview of the roadmap along with what’s in the pipeline for future development.

**What is the title of the blog post?**
Overview of security analytics in OpenSearch.

**Who are the authors?**
Jimish Shah

**What is the proposed posting date?**
March 2023

### PR DESCRIPTION
**Pull request by cwillum**
### Description
Blog post for Security Analytics introduction and overview
 
### Issues Resolved
Fixes #1221 

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

**Comment by cwillum** [comment]
@opensearch-project/security-analytics Please have one or two people review this blog draft for technical accuracy. Thanks.

File: _posts/2023-03-13-introducing-security-analytics.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+---
+layout: post
+title:  "Introducing Security Analytics"
+authors:
+  - christophermoore
+date:   2023-03-13 12:15:00 -0700
+categories:
+  - releases
+meta_keywords: security analytics
+meta_description: Provides and overview of Security Analytics and how it fits in with the OpenSearch Project
```
Comment (pajuric) [inline _posts/2023-03-13-introducing-security-analytics.md:10]: Update to meta keywords and description: 

**meta_keywords:** security analytics, security analytics in OpenSearch, open source SIEM tools

**meta_description:** Learn how the powerful security analytics capabilities in OpenSearch allow administrators to track and analyze activity in the cluster to combat cyber threats.

Comment (jimishs) [inline _posts/2023-03-13-introducing-security-analytics.md:10]: "an" ?

File: _posts/2023-03-13-introducing-security-analytics.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+---
+
+# Introducing Security Analytics
+
+In January of 2023, one of the largest wireless carriers in the US revealed that hackers had compromised a company database of customer information through unauthorized use of an API and had acquired the personal details of as many as 37 million individuals, including names, addresses, phone numbers, and account numbers. Once identified, the company was able to halt the malicious activity within a day. However, investigations indicated that the data breach had likely occurred at the end of November the previous year, giving the hackers over a month to extract the information.
```
Comment (jimishs) [inline _posts/2023-03-13-introducing-security-analytics.md:15]: Could we just say "One of the largest wireless carriers in the US revealed recently, that hackers had ..." . This leads better than with the date.

Comment (jimishs) [reply _posts/2023-03-13-introducing-security-analytics.md:15]: We can remove this line, we we've already drawn the attention of the reader. "However, investigations indicated that the data breach had likely occurred at the end of November the previous year, giving the hackers over a month to extract the information."

File: _posts/2023-03-13-introducing-security-analytics.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+Although unfortunate, occurrences like these now appear frequently in the media and have become almost commonplace in the modern world. With an ever-increasing volume of business data shared online and stored in the cloud, malicious attempts to disrupt companies’ business interests are now simply part of the reality of doing business. And in light of these threats, it’s therefore essential that companies have tools in place to detect and identify the signatures of possible attacks in real time so they can act to intervene as quickly as possible. To address these needs, OpenSearch now includes a package of pre-built features and functionality that allow businesses to quickly and easily roll out a security solution for confronting a full range of cyber threats. Security Analytics is this solution for OpenSearch, and it installs automatically with any distribution.
```
Comment (jimishs) [inline _posts/2023-03-13-introducing-security-analytics.md:17]: " ... in the media ..." doesnt support the core argument. i would consider editing the lead line here to something like "In the modern world where devices are interconnected across virtual boundaries, it has become commonplace for such security attacks to occur."

Comment (jimishs) [reply _posts/2023-03-13-introducing-security-analytics.md:17]: Should we say enterprises instead of business?

File: _posts/2023-03-13-introducing-security-analytics.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+## The role of Security Analytics in OpenSearch
+
+OpenSearch already includes powerful features that allow administrators to track and analyze activity within the cluster. They provide users the ability to identify anomalous behavior in system data and set up alerts and notifications to signal users when these events occur. However, scaling these existing tools to serve the specific needs of various industries, handling the volume of data they possess, and then assembling a team of professionals to manage and interpret the results, pose a number of challenges.
```
Comment (jimishs) [inline _posts/2023-03-13-introducing-security-analytics.md:21]: can you specify the type of activity? maybe say something like "analyze application logs and other security event logs" ?

Comment (jimishs) [reply _posts/2023-03-13-introducing-security-analytics.md:21]: I would not limit it to say "system data". can we make it generic

Comment (jimishs) [reply _posts/2023-03-13-introducing-security-analytics.md:21]: could we find another word in place of "industries" ? how about "security use cases" or "business use cases" ?

File: _posts/2023-03-13-introducing-security-analytics.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+Security Analytics takes a different approach. It lets businesses use their existing OpenSearch cluster for detecting security threats, rather than requiring them to bear the cost of duplicating and storing their data for a separate off-the-shelf security solution. Security Analytics also leverages a schema of rules that it applies to log data to look for matches with abnormal events. These standardized rules address specific threats known throughout the industry and form a knowledge base of known attacks. Combined with a number of robust features, the rules also allow Security Analytics to be highly customizable.
```
Comment (jimishs) [inline _posts/2023-03-13-introducing-security-analytics.md:23]: The significant cost savings isnt from using an existing OpenSearch cluster, but instead simply by using a single solution such as OpenSearch that provides a comprehensive suite of features and capabilities. In addition the lower cost and low latency performance at scale are important benefits that attract users to OpenSearch

Comment (jimishs) [reply _posts/2023-03-13-introducing-security-analytics.md:23]: "Schema of rules" isnt accurate. Instead we could say "leverages widely used Sigma rules to find threat patterns in your security event logs"

File: _posts/2023-03-13-introducing-security-analytics.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+## The parts of the whole
+
+To see how this works, let’s first look at some of the components in Security Analytics.
+
+### Detectors
+
+Central to Security Analytics are its detectors. Detectors are components that interpret log data, compare that data to a schema of rules, and define the triggers that generate findings and the alerts that notify administrators of possible security events in the system. Their advantage is that they offer flexibility through their configuration, and this allows a business to fine tune its detection capabilities for specific threats, wider objectives, or targets in between.
```
Comment (jimishs) [inline _posts/2023-03-13-introducing-security-analytics.md:31]: "in the system" -> "in your organization" or "in your infrastructure"
" and define the triggers" -> "enable you to configure alert triggers"

File: _posts/2023-03-13-introducing-security-analytics.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+### Rules
+
+Security Analytics leverages the versatility, portability, and up-to-date framework of [Sigma rules](https://github.com/SigmaHQ/sigma). These rules define the conditional logic applied to ingested log data and allow the system to identify an event of interest. In general, they determine exactly what type of event or attack the system is looking out for. Security Analytics uses these prepackaged, open-source rules as a starting point for describing relevant log events. But with their inherently flexible format and wide compatibility, users can import and customize Sigma rules to meet their needs.
```
Comment (jimishs) [inline _posts/2023-03-13-introducing-security-analytics.md:35]: "up-to-date" - > give we dont control Sigma, i would say "broadly covered" or something like that

Comment (jimishs) [reply _posts/2023-03-13-introducing-security-analytics.md:35]: i would remove phrases like "in general" or sentences ending in "for". It feels less confident and incomplete.

File: _posts/2023-03-13-introducing-security-analytics.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+### Log types
+
+Log data is key to understanding activity in any network. Whether it’s server logs, firewall logs, infrastructure logs, application logs, or endpoint logs, this data is elemental in gaining insight into the activity within a network. In terms of security coverage, the more log data a security infrastructure can process the better. Similarly, the wider the variety of log types that infrastructure can ingest, the more versatile that solution. Security Analytics launched with options for the most common log types. Since then, we have devoted significant energy to expanding support for more and more log types. And that trend will continue.
```
Comment (jimishs) [inline _posts/2023-03-13-introducing-security-analytics.md:39]: i would remove phrases like "more and more" and replace it with "more" or "additional"

File: _posts/2023-03-13-introducing-security-analytics.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+In terms of threat detection, the ability to correlate threat patterns across different log sources offers a clearer and more thorough view of potential malicious activity in a network. To enable Security Analytics to accomplish this, we have plans to create a Correlation engine that could not only draw these connections across log types but between current and previous findings.
+
+Other improvements on the horizon include chaining of alerts, better ingestion of log data, and further integration with other OpenSearch functionality. These are just a few of the steps along the way to making Security Analytics a highly effective tool for protecting your data and preventing disruptions to business. The modern reality of doing business may never be completely free from bad actors interested in unlawful access to and theft of other’s data. But the industry is responding to those circumstances in smart ways with smart solutions. In that spirit, Security Analytics addresses the challenges of keeping data safe from attacks so data owners can focus more time on the business of doing business.
```
Comment (jimishs) [inline _posts/2023-03-13-introducing-security-analytics.md:67]: i would rephrase "better ingestion"

File: _posts/2023-03-13-introducing-security-analytics.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+Closely related, we also plan to increase support of out-of-the-box dashboards for a larger pool of log types. Currently, when users create detectors for the **Network events**, **CloudTrail logs**, or **S3 access logs** log types, the system automatically generates a dashboard for visualizing the data generated by the detector. This will be true for more and more log types in future versions.
+
+In terms of threat detection, the ability to correlate threat patterns across different log sources offers a clearer and more thorough view of potential malicious activity in a network. To enable Security Analytics to accomplish this, we have plans to create a Correlation engine that could not only draw these connections across log types but between current and previous findings.
```
Comment (jimishs) [inline _posts/2023-03-13-introducing-security-analytics.md:65]: Feel free to link the correlation engine rfc in line. thanks

File: _posts/2023-03-13-introducing-security-analytics.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+
+## Looking ahead
+
+Since its release in version 2.4, Security Analytics has experienced refinements to both its functionality and number of features. However, there is room for improvement and more work to do before Security Analytics becomes the comprehensive tool we envisioned at the outset of development. Here are a few of the updates we hope to address in the near future.
+
+Keeping in line with our efforts to date, we will continue to expand log-type compatibility so that the options for using log data are as extensive as the systems that generate the data. This will be an incremental effort. But the list of supported log types will increase with each new version. Some log types we plan to add in upcoming releases include Cisco ASA, Palo Alto Networks Firewall logs, Cloudfront, SecurityHub findings, and Route53, to name just a few.
```
Comment (jimishs) [inline _posts/2023-03-13-introducing-security-analytics.md:61]: We should be careful in making this claim "But the list of supported log types will increase with each new version." i would revise it to what we wish to do rather than making any promises or setting the wrong expectations

Comment (jimishs) [reply _posts/2023-03-13-introducing-security-analytics.md:61]: I would rephrase this as this does not install confidence in the end user that this solution is ready for production ready environment and can detect potential threats - "Since its release in version 2.4, Security Analytics has experienced refinements to both its functionality and number of features. However, there is room for improvement and more work to do before Security Analytics becomes the comprehensive tool we envisioned at the outset of development. "

Comment (jimishs) [reply _posts/2023-03-13-introducing-security-analytics.md:61]: "Some log types we plan to add in upcoming releases include Cisco ASA, Palo Alto Networks Firewall logs, Cloudfront, SecurityHub findings, and Route53, to name just a few." 

while many of these are under discussion, i would not list them out here.

File: _posts/2023-03-13-introducing-security-analytics.md [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+
+### The user interface and dashboards
+
+The Security Analytics user interface is part of OpenSearch Dashboards and provides real-time visualizations for findings, alerts, and other analytics. Interaction with the Dashboards UI is intuitive and lets users easily navigate features and perform essential actions. Most importantly, the interface presents graphical information on network events happening in the moment. And to facilitate analysis of findings, we are continuously expanding support for dashboards that can present findings in a number of varied formats—such as charts, graphs, and maps—giving users the ability to customize their visualizations.
```
Comment (jimishs) [inline _posts/2023-03-13-introducing-security-analytics.md:43]: "The Security Analytics user interface is part of OpenSearch Dashboards"

This gives the impression that only Dashboards has support for this feature, when instead this can be entirely used with the rich set of APIs included.

**Comment by cwillum** [comment]
@jimishs Thanks for the thorough review. I've incorporated changes based on your comments. When you get a chance, could you have one more look? If it seems OK, I'll then move it through the documentation team review process. Thanks.
Also please add your personal information to the authors section.

File: _posts/2023-03-13-introducing-security-analytics.md [hunk 13]
```diff
@@ -0,0 +1,68 @@
+---
+layout: post
+title:  "Introducing Security Analytics"
+authors:
+  - christophermoore
+date:   2023-03-13 12:15:00 -0700
```
Comment (pajuric) [inline _posts/2023-03-13-introducing-security-analytics.md:6]: Please contact me to reschedule the release date for this blog.

Comment (pajuric) [reply _posts/2023-03-13-introducing-security-analytics.md:6]: @jimishs - when do you plan to push this live? I have tentatively scheduled it for 5/1/23.

Comment (pajuric) [reply _posts/2023-03-13-introducing-security-analytics.md:6]: @jimishs - Please advise on the schedule for completion for this blog. It's overdue on the publishing date.

**Comment by pajuric** [comment]
Approval to close from Jimish on 7/14/23
